.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,45 @@
+name: "CodeQL"
+
+on:
+  push:
+  pull_request:
+    branches: [ main ]
+    paths-ignore:
+      - '**.md'
+      - 'docs/**'
```

Comment:
nit: Maybe we do it this way:

```
  push:
    branches:
      - main
    paths-ignore:
      - 'docs/**'
  pull_request:
    paths-ignore:
      - 'docs/**'
```

then it will run on PR against all branch and not just main but push will be restricted to main branch only.

I took this sample from MVD.So we can do similar here.